File: 1.两数之和.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        //暴力
        vector<int> res;
         // for (int i = 0; i < nums.size();i++){
         //    for(int j = i + 1;j < nums.size();j++){
         //        if (nums[j] == target - nums[i]){
         //            res.push_back(i);
         //            res.push_back(j);
         //        }
         //    }
         // }
        // return res;
        unordered_map<int, int> m;
		for (int i = 0; i < nums.size(); i++)
		{
			if (m.find(target - nums[i]) != m.end()){
                res.push_back(m[target - nums[i]]);
                res.push_back(i);
                break;
            }
			m[nums[i]] = i;
		}
		return res;
    }
};

File: LeetCode 热题 HOT 100/1.两数之和.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
// 用数组 indexOf
var twoSum = function (nums, target) {
    let res = [];
    for (let i = 0; i < nums.length; ++i) {
        let idx = nums.indexOf(target - nums[i]);
        if (idx !== -1 && idx !== i) {
            res.push(i, idx);
            break;
        }
    }
    return res;
};

// 用map
var twoSum = function (nums, target) {
    let res = [];
    let map = new Map();
    for (let i = 0; i < nums.length; ++i) {
        let temp = target - nums[i];
        if (map.has(temp)) {
            res.push(i, map.get(temp));
            break;
        }
        map.set(nums[i], i);
    }
    return res;
};